master/internal/experiment.go

```diff
@@ -373,6 +376,23 @@ func (e *experiment) Receive(ctx *actor.Context) error {
 		}
 
 		ctx.Log().Info("experiment shut down successfully")
+
+	case *apiv1.ActivateExperimentRequest:
+		switch ok := e.updateState(ctx, model.ActiveState); ok {
```

Comment:
non-blocking: this is better as just an if-else given that this is just a bool variable.